### DISCUSSION TITLE
Create ISO 27701 referential

### DISCUSSION THREAD

**Discussion by danje57** [monarc-project/MonarcAppFO#427] [Q&A] [answered]
Dear all,

I've created the referential for ISO 27701, Privacy Information Management System.

However now, how to import it, and map it to ISO 27002:2013 and 2022 controls in Monarc?
[ISO27701 Monarc.xlsx](https://github.com/monarc-project/MonarcAppFO/files/8512699/ISO27701.Monarc.xlsx)

Regards,

**Comment by jerolomb** [comment]
Dear Danje57,

For that your Excel file has to be in the "MONARC" format. First have a 27701 in excel with column, code, label and category. 

After that go to the Knowledge Base, create the referential 27701. And after under the referential, click on the blue plus and import the Excel file. 

Once it's done, you have the referential. To link with other referential it's also to import an excel file for the mapping, you will find in the doc [1] the place in MONARC

Once you are on the right view you can download an example file. Fill it and upload it. Then the mapping is done. 

if you have any issue, don't hesitate

Jérôme 
[1] https://www.monarc.lu/documentation/user-guide/#referentials point 2

**Reply by danje57** [reply]
Thanks, this is the file now
[1.xlsx](https://github.com/monarc-project/MonarcAppFO/files/8513070/1.xlsx)

It can be imported into Monarc Training instance, but not in my SaaS environment :-/

Strange

**Reply by danje57** [reply]
On the SaaS,  Monarc display this; but I can't continue the import of the file
![Cap1](https://user-images.githubusercontent.com/16591723/164010472-a4c58e00-de1c-4eee-a921-8c3d03cafa3c.JPG)
![Cap2](https://user-images.githubusercontent.com/16591723/164010514-151902bf-9db7-4f20-a146-93e56b3fae57.JPG)

**Reply by jerolomb** [reply]
Seems it's an issue with ISO 27001:2013, you can start by importing first your referential and after via MOSP the iso 27002.

We will investigate

**Reply by danje57** [reply]
HEllo,

Finally I take time to finish the ISO 27701 Referential with match with new ISO 27002:2022 controls.

I would like to share it with you.

![image](https://user-images.githubusercontent.com/16591723/169097901-cbf630cd-5ee3-425d-82a4-2af84b725b42.png)


However, when I export my ISO27701 referential, it seems that the match is not saved in the exported CSV...

**Reply by danje57** [reply]
After search,
I can't find how to export ISO27701 referential with match to ISO 27002:2022.

I wish to share this with the community, however excepted in the GUI, I don't see the match in the exported file...

**Comment by jerolomb** [comment, marked as answer]
Hello danje57

One you have done the matching. You can click here : 

![image](https://user-images.githubusercontent.com/18661517/169473752-1e16cd3a-0c6a-4b74-9e01-6aac2b996c4f.png)

Let us know if it doesn't work

Have a nice day,

Jérôme

**Reply by danje57** [reply]
[ISO27701-ISO27002 2022 matchReferentials.csv](https://github.com/monarc-project/MonarcAppFO/files/8741077/ISO27701-ISO27002.2022.matchReferentials.csv)
[allControls_ISO 27701.csv](https://github.com/monarc-project/MonarcAppFO/files/8741078/allControls_ISO.27701.csv)

**Reply by danje57** [reply]
Here it is!

**Reply by danje57** [reply]
Hi Jerôme, I would like to share this as an object on MOSP.
But I don't see the option in Monarc as it exists for Monarc object for example.

However CSV can be used using Monarc's standard option.

Thanks, again for your help.

Jeremy

**Reply by cedricbonhomme** [reply]
Hi Jeremy,

Thank you for this. It's a really nice work !

I see the issue now, this kind of object is not exported to JSON from MONARC.
I will translate the referential (and the matching) to JSON thanks to script (similar to [this](https://github.com/CASES-LU/PyMOSP/blob/master/old/csv_to_mosp.py)). Then if you want I can give it to you, and you can put it on MOSP. Or I put it on MOSP with you as an author of course.

I can do this after tomorrow.

**Reply by danje57** [reply]
Thanks for the reply Cedric,

No problem for me it is for the community, so you export the JSON file on MOSP yourself no problem for me.

I'll probably do the same for ISO 27017 for the end of this month, I you have translation instruction I could do it myself, if not I'll post it on github.

Thanks for your help!

**Reply by cedricbonhomme** [reply]
@jfrocha kindly did it:

- https://objects.monarc.lu/object/view/5232
- https://objects.monarc.lu/object/view/5230

;-)

**Reply by danje57** [reply]
Thanks teams,

I started ISO 27017, however it's a little bit bigger :)

Have a nice day!

**Reply by danje57** [reply]
@cedricbonhomme and @jfrocha  as promised, I created ISO 27017 for cloud management with match with 27002:2022 controls.

Feel free to share it ;-)

[allControls_ISO 27017.csv](https://github.com/monarc-project/MonarcAppFO/files/9160123/allControls_ISO.27017.csv)
[matchReferentials.csv](https://github.com/monarc-project/MonarcAppFO/files/9160124/matchReferentials.csv)

**Reply by cedricbonhomme** [reply]
thanks again for this. It's very nice. I'll add it !

**Reply by cedricbonhomme** [reply]
For the moment the object is already here:
https://objects.monarc.lu/object/view/5233

thanks again !

I've used PyMOSP to push it. [Just for your information](https://www.monarc.lu/documentation/MOSP-documentation/api-v2.html#create-a-new-object). The comple code with the creation of the JSON:

```python
#! /usr/bin/env python
# -*- coding: utf-8 -

import csv

import pymosp
from config import CONFIG

CSV_FILE = 'allControls_ISO.27017.csv'
VALIDATING_SCHEMA = 12
OWNING_ORGANIZATION = 16

if __name__ == '__main__':
    # Construction of the object
    new_object = {
        "name": "ISO 27017",
        "description": "ISO 27017 for Cloud Management",
        "licenses": [{"license_id": "CC0-1.0"}],
        "schema_id": VALIDATING_SCHEMA,
        "org_id": OWNING_ORGANIZATION,
    }
    objects = {
        "uuid": "9a73a15e-bb36-434e-a2fe-c3ac1b7ed9a2",
        "label": "ISO 27017",
        "language": "EN",
        "authors": [
            "<Author-Name>"
        ],
        "refs": "https://www.iso.org/fr/standard/43757.html",
        "version": 1,
        "values": []
    }
    with open(CSV_FILE, newline='') as csvfile:
        reader = csv.reader(csvfile, delimiter=';')
        for row in reader:
            objects["values"].append({
                "uuid": row[0],
                "label": row[2],
                "category": row[3],
                "referential": "9a73a15e-bb36-434e-a2fe-c3ac1b7ed9a2",
                "referential_label": "ISO 27017",
                "code": row[1],
            })
    new_object["json_object"] = objects

    # Create the new object on MOSP
    token = CONFIG["TOKEN"]
    mosp = pymosp.PyMOSP("https://objects.monarc.lu/api/v2/", token)
    r = mosp.add_objects([new_object])
```

**Reply by danje57** [reply]
Thanks Cedric!

I'll probably make the PCI-DSS v4 in the future
But not now maybe on December.

I keep a track of the script for for sharing!

**Reply by cedricbonhomme** [reply]
OK, thank you. Do not worry for the push, we can help.

**Comment by danje57** [comment]
Thanks Jerome,

Cleaning my browser cache solve the issue and the arrow appears.

Now I'll see how to share the files with you.

Next week if I've time, I'll do the same for ISO 27017.
